x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/asset_inventory_search_bar.tsx

```diff
@@ -39,6 +41,18 @@ export const AssetInventorySearchBar = ({
     },
   } = useKibana().services;
 
+  const queryClient = useQueryClient();
+
+  const handleRefresh = () => {
+    queryClient.invalidateQueries({
+      queryKey: [QUERY_KEY_ASSET_INVENTORY],
+    });
+  };
+
+  const isFetchingData = useIsFetching({
+    queryKey: [QUERY_KEY_ASSET_INVENTORY],
+  });
```

Comment:
What about extracting this logic into a hook in the [cloud security package hooks folder](https://github.com/elastic/kibana/tree/main/x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/hooks) to be reused?


I.e how to consume it:

```js
const { refresh, isRefreshing } = useRefresh(QUERY_KEY_ASSET_INVENTORY);
```